features/networking/egress-ip.feature

```diff
@@ -232,3 +232,126 @@ Feature: Egress IP related features
     Then the step should succeed
     And the output should not contain "reg0=0x"
     """
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-19973
+  @admin
+  @destructive
+  Scenario: The egressIP should still work fine after the node or network service restarted
+    Given the valid egress IP is added to the node
+    And I have a project
+    And I have a pod-for-ping in the project
+
+    # add the egress ip to the project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ip %>"]} |
+
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | ifconfig.me |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+
+    # Reboot the node which patched egressIP
+    Given the host is rebooted and I wait it up to 600 seconds to become available
+    When I execute on the pod:
+      | curl | -s | --connect-timeout | 5 | ifconfig.me |
+    Then the step should succeed
+    And the output should contain "<%= cb.valid_ip %>"
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-15998
+  @admin
+  Scenario Outline: Invalid egressIP should not be acceptable
+    Given I select a random node's host
+    When I run the :patch admin command with:
+      | resource      | hostsubnet                     |
+      | resource_name | <%= node.name %>               |
+      | p             | {"egressIPs": <invalid_ip> }   |
+      | type          | merge                          |
+    Then the step should fail
+    And the output should contain "Invalid JSON Patch"
+
+    Examples:
+      | invalid_ip              |
+      | fe80::5054:ff:fedd:3698 |
+      | a.b.c.d                 |
+      | 10.10.10.-1             |
+      | 10.0.0.1/64             |
+      | 10.1.1/24               |
```

Comment:
Slightly off-topic but have we ever tested the alternate IPv4 representations?

> For example, the quad-dotted IP address 192.0.2.235 represents the 32-bit decimal number 3221226219, which in hexadecimal format is 0xC00002EB. This may also be expressed in dotted hex format as 0xC0.0x00.0x02.0xEB, or with octal byte values as 0300.0000.0002.0353.

https://en.wikipedia.org/wiki/IPv4#Address_representations

Reply:
It  doesn't support other format of IP address, it will show the warning  " egressIPs in body should match '^(([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])$",  I can add one example data here. Thanks Ross.